src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -830,6 +835,55 @@ public function isAbstract()
     }
 
     /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore.
+     *
+     * @param bool   $boolean
+     * @param string $template Template message to use if the definition is deprecated
+     *
+     * @return Definition the current instance
+     *
+     * @api
+     */
+    public function setDeprecated($boolean = true, $template = null)
```

Comment:
I copied the `setAbstract`, `setPublic` definitions but this is fine too, as there are two parameters